gallery_dl/extractor/discord.py

```diff
@@ -0,0 +1,370 @@
+# -*- coding: utf-8 -*-
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 2 as
+# published by the Free Software Foundation.
+
+"""Extractors for https://discord.com/"""
+
+from .common import Extractor, Message
+from .. import text, exception
+
+
+BASE_PATTERN = r"(?:https?://)?discord\.com"
+
+
+class DiscordExtractor(Extractor):
+    """Base class for Discord extractors"""
+    category = "discord"
+    root = "https://discord.com"
+    filename_fmt = "{message_id}_{num:>02}_{filename}.{extension}"
+    archive_fmt = "{message_id}_{num}"
+
+    cdn_fmt = "https://cdn.discordapp.com/{}/{}/{}.png?size=4096"
+
+    server_metadata = {}
+    server_channels_metadata = {}
+
+    def _init(self):
+        self.token = self.config("token")
+        self.enabled_embeds = self.config("embeds", ["image", "gifv", "video"])
+        self.enabled_threads = self.config("threads", True)
+        self.api = DiscordAPI(self)
+
+    def extract_message_text(self, message):
+        text_content = [message["content"]]
+
+        for embed in message["embeds"]:
+            if embed["type"] == "rich":
+                text_content.append(embed.get("author", {}).get("name", ""))
```

Comment:
```suggestion
                try:
                    text_content.append(embed["author"]["name"])
                except Exception:
                    pass
```

Doesn't construct a useless dict every time. Empty fields would get filtered out anyway.